packages/@lwc/engine-core/src/framework/decorators/register.ts

```diff
@@ -193,7 +212,15 @@ export function registerDecorators(
                 if (process.env.NODE_ENV !== 'production') {
                     validateFieldDecoratedWithApi(Ctor, fieldName, descriptor);
                 }
-                descriptor = createPublicPropertyDescriptor(fieldName);
+
+                // [W-9927596] If a component has both a public property and a private setter/getter
+                // with the same name, the property is defined as a public accessor. This branch is
+                // only here for backward compatibility reasons.
+                if (!isUndefined(descriptor) && !isUndefined(descriptor.get)) {
```

Comment:
Is it worth covering the case where there is only a `set`, but no `get`?

```suggestion
                if (!isUndefined(descriptor) && (!isUndefined(descriptor.get) || !isUndefined(descriptor.set)) {
```